docs/developer-guide/ksqldb-reference/select-pull-query.md

```diff
@@ -14,98 +14,130 @@ Synopsis
+the `WHERE` clause altogether. Table scans can be turned on for pull queries running in the 
+current CLI session with the command `SET 'ksql.query.pull.table.scan.enabled'='true';`. 
+They can also be enabled by default by setting a server configuration property with 
+`ksql.query.pull.table.scan.enabled=true`.Once table scans are enabled, the following 
+additional expressions are allowed:
+-   Key column(s) using range comparisons to literals.
+-   Non key columns to be used alone, without key references.
+-   Columns to be compared to other columns.
+-   References to subsets of columns from a multi-column key.
+-   Complex expressions without direct column references (e.g. instr(NAME_COL, 'hello') > 0).
 
+!!! note
+	Table scan based queries are just the next incremental step for ksqlDB pull queries. 
+	In future releases, we will continue pushing the envelope of new query capabilities and 
+	greater performance and efficiency.
 
+Examples
+--------
+Pull queries against a table `TOP_TEN_RANKS` created by using a 
+[CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statement:
+First, create a table named `GRADES` by using a [CREATE TABLE](../../ksqldb-reference/create-table) 
+statement:
 ```sql
 CREATE TABLE GRADES (ID INT PRIMARY KEY, GRADE STRING, RANK INT) 
-  WITH (kafka_topic = 'test_topic', value_format = 'JSON', partitions = 1);
+  WITH (kafka_topic = 'test_topic', value_format = 'JSON', partitions = 4);
 ```
-Create a derived table, named 
-`TOP_TEN_RANKS`, by using a [CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statement:
-
- ```sql
+Then, create a derived table named `TOP_TEN_RANKS` by using a 
+[CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statement:
+```sql
 CREATE TABLE TOP_TEN_RANKS 
   AS SELECT ID, RANK 
   FROM GRADES 
   WHERE RANK <= 10;
- ```
-You can fetch the current state of your materialized view, which is
-the `TOP_TEN_RANKS` derived table, by using a pull query:
-
+```
+If you want to look up only the student with `ID = 5` in the `TOP_TEN_RANKS` table using a pull query:
+```sql
+SELECT * FROM TOP_TEN_RANKS
+  WHERE ID = 5;
+```
+After enabling table scans, you can fetch the current state of your `TOP_TEN_RANKS` table using a pull query:
 ```sql
 SELECT * FROM TOP_TEN_RANKS;
 ```
-The following statement looks up only the student with `ID = 5` in the derived table:
-
+If you want to look up the students whose ranks lie in the range `(4, 8)`:
 ```sql
 SELECT * FROM TOP_TEN_RANKS
-  WHERE ID = 5;
+  WHERE ID > 4 AND ID < 8;
 ```
-!!! note
-	Currently, tables derived using Table-Table joins aren't queryable directly. To derive a queryable table, you can do:  
-	`CREATE TABLE QUERYABLE_JOIN_TABLE AS SELECT * FROM JOIN_TABLE;` and then issue pull queries against `QUERYABLE_JOIN_TABLE`.
-	
+
+Pull queries against a table `INNER_JOIN` that is created by joining multiple tables:

```

Comment:
I'm not entirely sure this example is helpful (the windowed one below is helpful because it shows the usage of window bounds on key lookups)

Reply:
We have stated in our 0.17 blog and 0.17 docs explicitly that Pull queries are not supported on Table-Table joins. This just clarifies to a certain degree that they are now supported.

Reply:
IMO I think it's sufficiently clarified above when you discuss that any CTAS table can be queried. Not a big deal though, we can keep it or get another opinion on it.

Reply:
Let's keep it. It is also one of the examples in our 0.18 blog